addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -54,7 +54,7 @@ var LinkDialog = Dialog.extend({
         if (this.data.range) {
             this.needLabel = selection.isCollapsed;
 
-            const $link = $(this.data.range.startContainer).filter("a");
+            const $link = $(this.data.range.startContainer).filter("a") || $(this.data.range.startContainer).closest("a");
```

Comment:
`link_tools` uses a util from `odoo-editor`: `getInSelection(editorDocument, 'a')`. Maybe we should use that here too and have the exact same behavior between the two. It behaves a little bit differently but I believe it's better (will find a link wherever in the selection, not just if it's an ancestor of the start container).

Reply:
Does `getInSelection` also selects link that wrap the selection entirely? Because this was what I was looking for with that call.

Reply:
```js
export function getInSelection(document, selector) {
    const selection = document.defaultView.getSelection();
    const range = !!selection.rangeCount && selection.getRangeAt(0);
    return (
        range &&
        (closestElement(range.startContainer, selector) ||
            [
                ...closestElement(range.commonAncestorContainer).querySelectorAll(selector),
            ].find(node => range.intersectsNode(node)))
    );
}
```
The first part of the condition is basically what you are doing with `filter || closest` (I don't think you need filter if you do closest). The second part will look at all the nodes in the selection for an anchor (this was written before we had `getSelectedNodes` but it could have used it instead of this strange contraption). Anyway, the first part should find links that wrap the selection entirely, in the same way that `closest` would.